### PR TITLE
Adding new bulk processor for elasticsearch implementation

### PR DESCRIPTION
**Pull request by HuiJun**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -7,4 +7,4 @@ springSecurityVersion=5.3.1.RELEASE
 springDataVersion=2.2.6.RELEASE
 jacksonVersion=2.10.3
 log4jVersion=2.13.1
-elasticVersion=7.1.1
+elasticVersion=7.9.0
```
Comment (dlamoris) [inline gradle.properties]: is there a 7.7.x version? the highest on aws right now is 7.7 and the client are forward but not backwards compatible

File: twc/src/main/java/org/openmbee/sdvc/twc/services/TwcRevisionMmsCommitMapService.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ public void setBranchRepository(BranchDAO branchRepository) {
      */
     public CommitsResponse updateTwcRevisionID(String projectId, String commitId, String revisionId) {
         CommitsResponse commitsResponse = new CommitsResponse();
-        if (revisionId.isEmpty() || revisionId.isBlank()) {
+        if (revisionId.isEmpty()) {
```
Comment (dlamoris) [inline twc/src/main/java/org/openmbee/sdvc/twc/services/TwcRevisionMmsCommitMapService.java]: should check for null first also

File: elastic/src/main/java/org/openmbee/sdvc/elastic/BaseElasticDAOImpl.java [hunk 3]
```diff
@@ -174,4 +189,35 @@ public E update(String index, BaseJson json) {
...
+
+    private static BulkProcessor getBulkProcessor(RestHighLevelClient client,  BulkProcessor.Listener listener) {
+        if (listener == null) {
+            listener = new BulkProcessor.Listener() {
+                @Override
+                public void beforeBulk(long executionId, BulkRequest request) {
+                }
+
+                @Override
+                public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
```
Comment (dlamoris) [inline elastic/src/main/java/org/openmbee/sdvc/elastic/BaseElasticDAOImpl.java:205]: should log any failures for now

File: elastic/src/main/java/org/openmbee/sdvc/elastic/BaseElasticDAOImpl.java [hunk 4]
```diff
@@ -31,6 +41,8 @@
 
 public abstract class BaseElasticDAOImpl<E extends Map<String, Object>> {
 
+    private final Logger logger = LogManager.getLogger(getClass());
```
Comment (dlamoris) [inline elastic/src/main/java/org/openmbee/sdvc/elastic/BaseElasticDAOImpl.java]: use slf4j LoggerFactory.getLogger(getClass())

File: example/example.gradle [hunk 5]
```diff
@@ -24,7 +24,6 @@ dependencies {
         project(':jupyter'),
         project(':permissions'),
         project(':webhooks'),
-        project(':twc'),
```
Comment (dlamoris) [inline example/example.gradle:27]: take out ldap too

**Review by dlamoris — CHANGES_REQUESTED** [review]

File: example/example.gradle (hunk 5 above)
Comment (HuiJun) [reply example/example.gradle:27]: done

File: twc/src/main/java/org/openmbee/sdvc/twc/services/TwcRevisionMmsCommitMapService.java [hunk 6]
```diff
@@ -36,7 +36,7 @@ public void setBranchRepository(BranchDAO branchRepository) {
      */
     public CommitsResponse updateTwcRevisionID(String projectId, String commitId, String revisionId) {
         CommitsResponse commitsResponse = new CommitsResponse();
-        if (revisionId.isEmpty() || revisionId.isBlank()) {
+        if (revisionId != null && revisionId.isEmpty()) {
```
Comment (dlamoris) [inline twc/src/main/java/org/openmbee/sdvc/twc/services/TwcRevisionMmsCommitMapService.java]: should be == null ||

**Review by dlamoris — APPROVED** [review]

**Pull request merged**